lib/daru/dataframe.rb

```diff
@@ -1578,6 +1578,10 @@ def pivot_table opts={}
       raise ArgumentError,
         "Specify grouping index" if !opts[:index] or opts[:index].empty?
 
+      arr = []
+      self.size.times {|i| arr[i] = i}
+      new_dataframe_index = Daru::Index.new(arr)
+      self.index = new_dataframe_index
```

Comment:
You're changing the DataFrame interally by re-assigning `self.index`. This is unexpected behaviour for `pivot_table`.